src/app/api/webhook/webhook.service.ts

```diff
@@ -121,16 +121,31 @@ class WebhookService extends BaseService {
   }
 
   async handleUserDeleted(assigneeId: string, assigneeType: AssigneeType) {
-    if (assigneeType === AssigneeType.company) return
-
     const tasksService = new TasksService(this.user)
     // Delete corresponding tasks
     console.info(`Deleting all tasks for ${assigneeType} ${assigneeId}`)
-    await tasksService.deleteAllAssigneeTasks(assigneeId, assigneeType)
+    const tasks = await tasksService.deleteAllAssigneeTasks(assigneeId, assigneeType)
 
-    // So how do we delete corresponding notifications, you ask? Good question.
-    // Copilot calls multiple `client.updated` when unassigning clients companies after `company.deleted` is called.
-    // So handling deleting company task notifications should be done in `client.updated` company unassignment logic instead
+    // Now delete any and all associated notifications triggered on behalf of company tasks for clients.
+    // i.e. decrement client task count in CU portal
+    if (assigneeType === AssigneeType.company) {
+      const deletedTaskIds = tasks.map((task) => task.id)
+      await this.handleDeletingTaskNotifications(deletedTaskIds)
+    }
+  }
+
+  private async handleDeletingTaskNotifications(taskIds: string[]) {
+    const clientTaskNotifications = await this.db.clientNotification.findMany({
```

Comment:
is `db.clientNotification` created to seprately store client notifications and not IU notifications?

Reply:
Yes